src/libcore/iter.rs

```diff
@@ -3030,6 +3030,45 @@ pub fn repeat<T: Clone>(elt: T) -> Repeat<T> {
     Repeat{element: elt}
 }
 
+/// An iterator that yields nothing.
+#[unstable(feature="core", reason = "new addition")]
+pub struct Empty<T>(marker::PhantomData<T>);
+
+impl<T> Iterator for Empty<T> {
+    type Item = T;
+
+    fn next(&mut self) -> Option<T> {
+        None
+    }
+}
+
+/// Creates an iterator that yields nothing.
+#[unstable(feature="core", reason = "new addition")]
+pub fn empty<T>() -> Empty<T> {
+    Empty(marker::PhantomData)
+}
+
+
+/// An iterator that yields an element exactly once.
+#[unstable(feature="core", reason = "new addition")]
+pub struct Once<T> {
+    inner: ::option::IntoIter<T>
```

Comment:
`::option::IntoIter<T>` isn't really necessary. You could just use `Option` + `take()`.